scripts/project_detection.sh

```diff
@@ -11,7 +11,16 @@
 
 # Project detection
 read_project_name_file () {
-    PROJECT_NAME_FILE='../../scripts/project_name.txt'
+    SCRIPT_DIR=$(pwd)
+
+    PROJECT_NAME_FILE="scripts/project_name.txt"
+
+    if echo "$SCRIPT_DIR" | grep -q "/framework/scripts" || echo "$SCRIPT_DIR" | grep -q "/tests/scripts"; then
```

Comment:
Seems unnecessary if we assume the scripts are always invoked from the project's root (which as I said on the other PR, is a standard assumption - too bad it's often unwritten).